src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -352,8 +357,11 @@ export const useSearch = (services: DiscoverViewServices) => {
         }
       }
 
-      filterManager.setAppFilters(actualFilters);
-      data.query.queryString.setQuery(query);
+      // Filters in URL are higher priority than the filters in saved search
+      const urlFilters = (osdUrlStateStorage.get('_q') as QueryState)?.filters ?? [];
+      if (!urlFilters || urlFilters.length === 0) {
+        filterManager.setAppFilters(actualFilters);
```

Comment:
wouldn't we want to make sure to call `filterManager.setAppFilters([]);` when `urlFilters` is bad?